sourcepolicy/sourcepolicy.go

```diff
@@ -0,0 +1,28 @@
+// Package sourcepolicy provides the source policy types
+package sourcepolicy
+
+import binfotypes "github.com/moby/buildkit/util/buildinfo/types"
+
+// SourceType defines the source type.
+type SourceType = binfotypes.SourceType
+
+// Source is a subset of github.com/moby/buildkit/util/buildinfo/types.Source .
+type Source struct {
+	// Type defines the SourceType source type (docker-image, git, http).
+	Type SourceType `json:"type,omitempty"`
+	// Ref is the reference of the source.
+	Ref string `json:"ref,omitempty"`
+	// Pin is the source digest.
+	Pin string `json:"pin,omitempty"`
+}
+
+// SourcePolicy provides pinning information to ensure determinism of sources.
+type SourcePolicy struct {
+	// Version is the version of the SourcePolicy structure.
+	// Must be set to 1 currently.
+	Version int `json:"version"`
```

Comment:
Is there any reason we couldn't use the capability-style checking that we have for versioning things today? We don't have explicit version numbers in other places I don't think, and capabilities would let us have a fine-grained control over small differences in the API over time.

Reply:
@tonistiigi WDYT?

Reply:
This is true for API controls. But if we imagine that in the future user would have a way to write a policy into some file, then if client that converts it into this policy object would ignore things it does not understand. Even if the server understands these fields client does not know what caps to request from the daemon.

This could be possibly implemented as user writing a version into a file and the client just converting that version to a `CAP_POLICY_VERSION_N` before sending. Then a special field in here is not needed.

Reply:
> This could be possibly implemented as user writing a version into a file

What file?

Reply:
Not defined atm but there needs to be some way user can pass this that is converted to `SourcePolicy` struct. Read also the points about frontend integration that would probably also use the same file.

Reply:
Removed the version field and added `CapSourcePolicy`